src/util/mapFolderData.js

```diff
@@ -3,25 +3,50 @@ import { getElapsedTime } from "./getElapsedTime";
 
 export const mapFolderData = (folder) => {
   if (!folder) return [];
-  const { name, owner, links } = folder;
+  const { name, owner, data, links, folder_id } = folder;
 
-  const mapLinks = (link) => {
-    const { id, createdAt, url, imageSource, title, description } = link;
-    return {
-      id,
-      url,
-      imageSource,
-      alt: `${title ?? url}의 대표 이미지`,
-      elapsedTime: getElapsedTime(createdAt),
-      description,
-      createdAt: format(new Date(createdAt), "yyyy. MM. dd"),
-    };
+  const mapLinks = (data) => {
+    if (!links) {
+      const {
+        id,
+        name,
+        created_at,
+        url,
+        image_source,
+        folder_id,
+        title,
+        description,
+      } = data;
+      return {
+        id,
+        url,
+        name,
+        imageSource: image_source,
+        folder_id,
+        alt: `${title ?? url}의 대표 이미지`,
+        elapsedTime: getElapsedTime(created_at),
+        description,
+        createdAt: format(new Date(created_at), "yyyy. MM. dd"),
+      };
+    } else {
+      const { id, createdAt, url, imageSource, title, description } = data;
+      return {
+        id,
+        url,
+        imageSource,
+        alt: `${title ?? url}의 대표 이미지`,
+        elapsedTime: getElapsedTime(createdAt),
+        description,
+        createdAt: format(new Date(), "yyyy. MM. dd"),
+      };
+    }
```

Comment:
```suggestion
    const {
      id,
      name,
      created_at,
      url,
      image_source,
      folder_id,
      title,
      description,
      
      createdAt,
      imageSource,
    } = data;

    if (!links) {
      return {
        id,
        url,
        name,
        imageSource: image_source,
        folder_id,
        alt: `${title ?? url}의 대표 이미지`,
        elapsedTime: getElapsedTime(created_at),
        description,
        createdAt: format(new Date(created_at), "yyyy. MM. dd"),
      };
    } else {
      return {
        id,
        url,
        imageSource,
        alt: `${title ?? url}의 대표 이미지`,
        elapsedTime: getElapsedTime(createdAt),
        description,
        createdAt: format(new Date(), "yyyy. MM. dd"),
      };
    }
```

이런식으로 블록문 바깥에서 디스트럭처링 할당을 해두고, 가져다 쓰면 하나의 값으로 관리할 수 있게 됩니다.

+ 질문 - createdAt 과 created_at, imageSource와 image_source의 차이가 뭘까요?
- 지금으로서는 같은 값이지만 단순히 네이밍 컨벤션을 맞추지 않은 문제 같은데 맞을까요?

Reply:
넵, 서버에서 오는 데이터의 이름이 달라서 따로 지정해두었습니다.